### DISCUSSION TITLE
Welcome to lua-protobuf Discussions!

### DISCUSSION THREAD

**Discussion by starwing** [starwing/lua-protobuf#169] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by NDR008** [comment]
I'm totally new to protobuf, but got myself through some very basic tutorials on python.
Now I want to try and get things working in lua.

Example in python, I made game.proto, and then used protoc to generate game_pb2.py
```py
import game_pb2 as Game

myGame = Game.observation()

print(myGame.debug, myGame.vehicle.current_brake)
```

after loading the protobuf in lua,
How can I instantiate an object of the type observation in lua?

**Reply by starwing** [reply]
Lua has not `class`, and lua-protobuf do not support "generate the message class" like python or C++.

You only need create table, with the same field with messages. and call `pb.encode()` (or get this table by call `pb.decode()`)

**Reply by NDR008** [reply]
Got it working :) Finally understood how to use it.
